presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcPreparedStatement.java

```diff
@@ -101,6 +103,46 @@ public void testExecuteQuery()
         }
     }
 
+    @Test
+    public void testGetMetadata()
+            throws Exception
+    {
+        try (Connection connection = createConnection("blackhole", "blackhole")) {
+            try (Statement statement = connection.createStatement()) {
+                statement.execute("CREATE TABLE test_get_metadata (" +
+                        "c_boolean boolean, " +
+                        "c_decimal decimal, " +
+                        "c_decimal_2 decimal(10,3)," +
+                        "c_varchar varchar, " +
+                        "c_varchar_2 varchar(10), " +
+                        "c_row row(x integer, y array(integer)), " +
+                        "c_array array(integer), " +
+                        "c_map map(integer, integer))");
+            }
+
+            List<String> colNames = Arrays.asList("c_boolean", "c_decimal", "c_decimal_2",
+                    "c_varchar", "c_varchar_2", "c_row", "c_array", "c_map");
+            List<String> expectColTypes = Arrays.asList("boolean", "decimal(38,0)", "decimal(10,3)",
+                    String.format("varchar(%s)", Integer.MAX_VALUE), "varchar(10)", "row()", "array", "map");
```

Comment:
the type name of a `row` type is `row()` due to legacy code https://github.com/prestosql/presto/blob/master/presto-client/src/main/java/io/prestosql/client/ClientTypeSignature.java#L161 will generate the type string using the arguments, noted that the generation function `rowToString` is `deprecated`.